MantidQt/CustomDialogs/src/StartLiveDataDialog.cpp

```diff
@@ -439,5 +466,60 @@ void StartLiveDataDialog::initListenerPropLayout(const QString &inst) {
   }
   ui.listenerProps->setVisible(true);
 }
+
+/**
+ * Slot to update list of available connections when instrument is changed
+ *
+ * @param inst_name Name of selected instrument
+ */
+void StartLiveDataDialog::updateConnectionChoices(const QString &inst_name) {
+  // Reset the connections listed
+  ui.cmbConnection->clear();
+  ui.cmbConnection->addItem("[Custom]");
+
+  // Add available LiveListenerInfo names based on selected instrument
+  const auto &inst =
+      ConfigService::Instance().getInstrument(inst_name.toStdString());
+  for (const auto &listener : inst.liveListenerInfoList()) {
+    ui.cmbConnection->addItem(QString::fromStdString(listener.name()));
+  }
+
+  // Select default item
+  auto selectName = QString::fromStdString(inst.liveListenerInfo().name());
+  auto index = ui.cmbConnection->findText(selectName);
+  ui.cmbConnection->setCurrentIndex(index);
+}
+
+/**
+ * Slot to update connection parameters when connection selected
+ *
+ * @param connection Name of selected live listener connection
+ */
+void StartLiveDataDialog::updateConnectionDetails(const QString &connection) {
+  // Custom connections just enable editting connection parameters
+  if (connection == "[Custom]") {
```

Comment:
Good, though I might also consider making ```[Custom]``` some kind of static const member to avoid future mistakes.